toolsrc/src/vcpkg/base/files.cpp

```diff
@@ -101,6 +101,63 @@ namespace vcpkg::Files
             return status_implementation(false, p, ec);
         }
 
+        fs::path read_symlink_implementation(const fs::path& oldpath, std::error_code& ec)
+        {
+#if defined(_WIN32) && !VCPKG_USE_STD_FILESYSTEM
+            auto handle = CreateFileW(oldpath.c_str(),
+                                      GENERIC_READ,
```

Comment:
```suggestion
                                      0, // open just the metadata
```

I think this should work, and it's not necessary to have read access on the final file.